src/main/kotlin/com/terraformation/backend/customer/api/ProjectsController.kt

```diff
@@ -180,3 +213,27 @@ data class UpdateProjectRequestPayload(
 data class GetProjectResponsePayload(val project: ProjectPayload) : SuccessResponsePayload
 
 data class ListProjectsResponsePayload(val projects: List<ProjectPayload>) : SuccessResponsePayload
+
+data class ProjectUserRoleResponsePayload(
+    val userId: UserId,
+    val email: String,
+    val firstName: String?,
+    val lastName: String?,
+    val role: ProjectInternalRole? = null,
+    val roleName: String? = null
+) {
+  constructor(
+      user: TerrawareUser,
+      projectInternalUser: ProjectInternalUsersRow
+  ) : this(
+      userId = user.userId,
+      email = user.email,
+      firstName = user.firstName,
+      lastName = user.lastName,
+      role = projectInternalUser.projectInternalRoleId,
+      roleName = projectInternalUser.roleName,
```

Comment:
Wasn't sure if response of both role and roleName was ok, or if I should combine them into one "role" with a getter

Reply:
This seems right to me. Combining them means the logic to figure out if it's an enum value or a custom one gets more complicated.